### ISSUE TITLE
Move away from php-imap extension

### ISSUE THREAD

**Issue by bobvandevijver** [secit-pl/imap-bundle#46]
The PHP-IMAP extension will be removed from PHP core with 8.4 (https://php.watch/versions/8.4/imap-unbundled). Maybe it is a good idea to migrate to https://github.com/Webklex/php-imap instead?

**Comment by mshannaq** [comment]
Yes  IMAP extension is no longer part of PHP 8.4 Core, which can still be [installed from PECL](https://pecl.php.net/package/imap) for Existing/legacy PHP applications thats depends on  IMAP extension.

**Comment by secit-pl** [comment]
I've thought about it but php-imap/php-imap and Webklex/php-imap are not compatible with each other and I'll be honest, I still have no idea how to implement Webklex/php-imap support without losing php-imap/php-imap support and big refactoring and bc-break. I also don't want to go into a wrapper for every possible library for imap support.

Any suggestions would be appreciated.

Currently my best idea is to create a separate bundle under this, but I'm not entirely sure this idea is a good one to maintain two separate repositories.

**Comment by ThomasLandauer** [comment]
IMO, `php-imap/php-imap` is dead. Reasons:
* It depends on `ext-imap`, i.e. it's riding a dead horse.
* It's unmaintained: The latest merging of a PR happened in December 2022.

As you say, supporting `Webklex/php-imap` *and* `php-imap/php-imap` makes no sense. So what about abandoning this bundle here, and starting from scratch with a new bundle just for `Webklex/php-imap` ?

**Comment by bobvandevijver** [comment]
I've ended up migrating my project to just `webklex/php-imap`, as a complete bundle with configuration was not necessary.

**Comment by garbast** [comment]
Maybe I can throw my hat in the ring.

I forked php-imap/php-image to evoweb/php-imap and started to refactor the package to modern standards. Almost all public API kept their signature. Only functions with _ in the name got renamed. So basically, by now it's a dropin replacement.
